### DISCUSSION TITLE
Does linkerd tap command detect non HTTP connections ?

### DISCUSSION THREAD

**Discussion by rohitsakala** [linkerd/linkerd2#6204] [Q&A] [answered]
I was trying to list TCP requests between two components in the service mesh using `linkerd tap` command but it is not showing anything. I could see HTTP requests using` linkerd tap` but no TCP connections (which are not HTTP).

Thanks

**Comment by kleimkuhler** [comment, marked as answer]
Yes tap only supports HTTP right now. It's core functionality is about showing the requests and responses for connections open on the target pod, but it also can additional metadata that is only associated with HTTP messages.

For TCP we don't really have this split between requests and responses and we would not be able to retrieve some of the additional metadata that we currently do. It may still be possible to support, but it would a feature that would require a lot of control plane and proxy work to support.

Is there a specific reason that you are asking or an issue that you are trying to solve that requires this?

**Reply by rohitsakala** [reply]
Thank you for the reply. My specific case is that an app pod in the service mesh is pushing a docker image to a private internal registry pod through kubernetes service and the `linkerd tap` method was not able to detect this. `linkered tap` was able to detect the requests for the pull docker image but this pull happens through http.

**Reply by kleimkuhler** [reply]
Okay yep so your thinking is correct on this and you won't see the TCP connection for pushing the image.
